### ISSUE TITLE
v8 installation issues: mongodb empty, wallet creation example still uses -f flag,...

### ISSUE THREAD

**Issue by educob** [bitpay/bitcore#1690]
I have installed bitcore v8 following the documentation:

```
npm install -g bitcore-node@beta
bitcore-node
```
Besides a couple of deprecated warning I get the following message: "info: Syncing from 0 to 377601 for chain BTC".

I see that in mongodb some collections have been created: blocks, coins, transactions, walletaddresses & wallets.
After hours the blocks collections is empty.

The example to create a wallet uses the old flag to create the wallet in a file:
`./bin/wallet-create "testing wallet creation" 1-1 --coin btc --network testnet -f ~/newtestwallet.dat`
Besides, what folder is bin located in?

The bitcore.config.json file sample is only for regtest so I have no idea how to configure it for the main network.

Any help would be appreciated. Documentation for v8 is almost non existent.

**Comment by micahriggan** [comment]
Try running it from this github repo. Been a while since we've pushed to npm.

If you're getting "Syncing from x to y" and it's not writing anything, make sure the node you're syncing from is fully synced. 

If the node you're syncing from isn't fully synced, then it won't respond to bitcore asking for headers or blocks :)

**Comment by educob** [comment]
Thanks for the help. 
Once the node synched, bitcore started popuulating mongodb.

But how can I create a wallet and addresses from javascript?
Thanks again.

**Comment by micahriggan** [comment]
I would refer to the files in packages/wallet-client for the scripts to do common things like deriving addresses and importing addresses etc. 

Eventually we'll get some nice docs and examples of how to do everything, but right now this is very much in flight

**Comment by educob** [comment]
Micahriggan thanks for answering.

There 2 packages with the word wallet:
1) bitcore-wallet. This is a Command Line Interface where wallets are saved in files.
2) bitcore-wallet-client. This is a javascript library where wallets are also saved in files.

In version v8 everything is saved in mongodb. I need to be able to create a wallet saved in mongodb.

Could you please help me?

Thanks.

**Comment by micahriggan** [comment]
https://github.com/bitpay/bitcore/tree/v8.0.0/packages/bitcore-client

This is where most of the V8 client examples are. The readme has some examples that should be fairly

**Comment by educob** [comment]
Thanks. 

I'll try to see if it saves the wallet on mongodb but I have taken a fast look and I see it uses levelDB.

**Comment by micahriggan** [comment]
The wallet file is stored client side with leveldb, that is for the private keys. Mongodb is server side and stores only the walletid and the addresses that the wallet controls.

**Comment by educob** [comment]
Thanks for the answer.
I have tried "import client from 'bitcore-client' and it says to run "npm install --save bitcoin-client". But this gives an error: "E404 Not Found: bitcore-client@latest".

There is no documentation at all. 
How can I install this library and find an example please!!!

Thanks.

**Comment by educob** [comment]
I have tried to install bitcore-client with variations of: "npm install --save https://github.com/bitpay/bitcore/tree/v8.0.0/packages/bitcore-client"

It doesn't work either.

How is this library installed?
Where can I find an example of wallet & address creation?
Thanks.

**Comment by educob** [comment]
After hours and hours lost trying to install the library I finally ask in npm.com forum and the answer is:

"You can’t install it using npm i, since this tree does not provide a valid package.json"

I am beginning to think that bitpay doesn't want people to use the library: no docu, no examples, no way to install the library using npm.

**Comment by skddc** [comment]
v8 is still beta, so it's likely best to install it from source. You can install packages locally by just using the path on the filesystem that leads to the `package.json`/directory. But if you're not prepared to hit bugs, then you're best advised to not use beta software, and either use the stable version or wait for the beta to become stable.

**Comment by educob** [comment]
Of course I am prepared to hit bugs but I am not an expert at npm (and many other things) and I would be grateful for a little more detail.

For instance I don't understand what you mean by "You can install packages locally by just using the path on the filesystem that leads to the package.json/directory".  You mean my filesystem? github filesystem? How in whatever case?
I have spent a lot of  hours with this issue and still haven't been able to install the library.

**Comment by skddc** [comment]
I see.

1. Download the this source code repository onto your computer. [You can do this by using Git on the command line, or some kind of graphical user interface, like e.g. GitHub for Mac or Windows.](https://help.github.com/articles/cloning-a-repository/)
2. Now that you have this repository on your computer, you can install any of the packages contained in it like so: `npm install [path/to/bitcore/packages/thepackageyouwant]` ([npm install help](https://docs.npmjs.com/cli/install))

**Comment by educob** [comment]
Thanks.

That made all the diffrerence.

**Comment by Fantic666** [comment]
Dear
may i ask if you got v8 to work for a full sync ?
What kind of BitcoinD you are running ?
Many thanks for your help.
Regards

**Comment by educob** [comment]
No. It didn't run at all. I have lost a whole week with bitcore (besides I asked in two bitcore forums how to create a wallet using javascript and no answer). I give up bitcore. They don't want outsiders.

**Comment by Fantic666** [comment]
thanks for your reply. was the full sync a problem? my sync speed goes down and down ...

**Comment by educob** [comment]
Synching was taking longer and longer. It took some time to sync a 0.001%.
A pity cause having the node in mongoDB allows for web reactivity.

**Comment by Fantic666** [comment]
crap - i give up at 66% ... 
thats worse - the performance seems to be outstanding compared to v5 
maybe because it didnt finished the sync - but my queries run 6 times faster :(
could someone else help ?
thanks

**Comment by skddc** [comment]
There was a rather large spam attack on testnet 3 recently, lasting weeks. When it gets to those blocks the sync slows down considerably. At least I think that's what I saw. Basically a few thousand blocks towards the very end.

**Comment by Fantic666** [comment]
many thanks for your reply 
i used the mainnet - is the testnet 10 times smaller ?
25GB - my bitcoind is around 250 GB and mongodb around 500GB
may i ask what config files you´re using for the bitcoind and bitcore-node ?

**Comment by educob** [comment]
I was loading the mainnet.

**Comment by micahriggan** [comment]
@Fantic666 500 GB sounds about right for mainnet. If your sync paused, may I ask if you pulled in recent commits? I added an index yesterday or the day before to coins. So if you pulled that in, that could be the issue. You'd have to run mongod from a terminal and see if you you see something like "indexing x%"

**Comment by Fantic666** [comment]
@micahriggan many thanks for your answer.
could you explain the complete steps ? 
the sync did not pause - it will be slower and slower and slower ...
i tried some other blockexplorer in c# and mysql before ... there i run into the same issues
for me it looks like the amount of data is to much - some queries need ages - thats my opinion.
Thanks for your time.
